bears/generate_package.py

```diff
@@ -159,12 +161,23 @@ def main():
                                       substitution_dict)
 
             bear_dist_name = bear_name + '-' + bear_version
-            if args.register:
+            if args.register and not args.bear:
                 perform_register(os.path.join('bears', 'upload', bear_name),
                                  bear_dist_name)
-            if args.upload:
+            if args.upload and not args.bear:
                 perform_upload(os.path.join('bears', 'upload', bear_name))
 
+    if args.bear:
```

Comment:
I had a better code as you said in previous commits, but wasnt covering full with travis.
So was trying to figure out how the tests work. Hence ended up with this.

Anyways, have fixed it in latest commit